File: gc-app/src/main/java/rocks/frick/gc/application/GetFullActivityService.java
package rocks.frick.gc.application;

import java.util.Optional;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import rocks.frick.gc.Ride;
import rocks.frick.gc.application.spi.ActivityRepository;

public class GetFullActivityService {

  private static final Logger log = LoggerFactory.getLogger(GetFullActivityService.class);

  private final ActivityRepository activityRepository;

  public GetFullActivityService(ActivityRepository activityRepository) {
    this.activityRepository = activityRepository;
  }

  public Optional<Ride> getFullWorkout(String id) {
    log.debug("Retrieving full workout for ID: {}", id);
    Ride workout = activityRepository.findById(id);

    return Optional.ofNullable(workout);
  }
}

File: gc-core/src/main/scala/gc/application/GetFullActivityService.scala
package gc.application

import scala.jdk.OptionConverters._

import com.typesafe.scalalogging.Logger
import gc.spi.FindActivityById
import gc.model.Activity

class  GetFullActivityService(activityRepository: FindActivityById) {
  private val log = Logger("GetFullActivityService")

  def getFullWorkout(id: String): Option[Activity] = {
    log.debug(s"[CORE] Retrieving full workout for ID: $id")
    activityRepository.findById(id)
  }
}
